test/Approximators/RangeApproximator/randsvd.jl

```diff
@@ -0,0 +1,334 @@
+        end
+
+        # Test constructor
+        let compressor = TestCompressor(),
+            power_its = 2,
+            rand_subspace = false,
+            rf = RandSVD(
+                compressor = compressor, 
+                power_its = power_its, 
+                rand_subspace = rand_subspace
+            ) 
+
+            @test typeof(rf.compressor) == TestCompressor
+            @test rf.power_its == 2
+            @test rf.rand_subspace == false
+        end
+    
+    end
+
+    @testset "RandSVD Recipe" begin
+        @test_range_approximator RandSVDRecipe
+        supertype(RandSVDRecipe) == ApproximatorRecipe
+
+        # test the fieldnames and types
+        @test fieldnames(RandSVDRecipe) == (
+            :n_rows, :n_cols, :compressor, :power_its, :rand_subspace, :U, :S, :V
+        )
+        @test fieldtypes(RandSVDRecipe) == (
+            Int64, 
+            Int64, 
+            CompressorRecipe, 
+            Int64, 
+            Bool, 
+            AbstractArray, 
+            AbstractVector, 
+            AbstractArray
+        )
+    end
+    
+    @testset "RandSVD: Complete Approximator" begin
+        # Test when correct cardinality, right, is specified
+        let n_rows = 10,
+            n_cols = 10,
+            compression_dim = 5,
+            cardinality = Right(),
+            A = rand(n_rows, n_cols)
+
+            compressor = TestCompressor(cardinality, compression_dim)
+            approx = RandSVD(compressor, 2, true)
+            approx_rec = complete_approximator(approx, A)
+            approx_rec.compressor.cardinality == Right()
+
+            @test approx_rec.power_its == 2
+            @test approx_rec.rand_subspace == true
+            @test approx_rec.n_rows == 10
+            @test approx_rec.n_cols == 5
+        end
+        
+        # Test when wrong cardinality, left, is specified
+        let n_rows = 10,
+            n_cols = 10,
+            compression_dim = 5,
+            cardinality = Left(),
+            A = rand(n_rows, n_cols)
+
+            compressor = TestCompressor(cardinality, compression_dim)
+            approx = RandSVD(compressor, 2, false)
+            @test_logs (:warn,
+                "Compressor with cardinality `Left` being applied from `Right`."
+            ) complete_approximator(approx, A)
+            approx_rec = complete_approximator(approx, A)
+            approx_rec.compressor.cardinality == Left()
+            
+            @test approx_rec.power_its == 2
+            @test approx_rec.rand_subspace == false 
+            @test approx_rec.n_rows == 10
+            @test approx_rec.n_cols == 5
+        end
+        
+    end
+
+    @testset "RandSVD Recipe: rapproximate" begin
+        # By testing the rapproximate function we also test rapproximate!
+        # with power iterations
+        let n_rows = 10,
+            n_cols = 10,
+            compression_dim = 5,
+            cardinality = Right(),
```

Comment:
The number of power iterations and whether orthogonalization is used should be passed as test parameters to the `let` block.